juju/client/connection.py

```diff
@@ -95,7 +95,7 @@ class Connection:
     @classmethod
     async def connect(
             cls,
-            endpoint,
+            endpoint=None,
```

Comment:
What's the point of this change if you then add run-time validation for it below?

If the concern is about passing an expanded dict for keyword args, you should still be able to do that with some positional params, as long as those are satisfied:

```python
In [1]: def foo(a, b=None):
   ...:     print(a, b)
   ...:     

In [2]: foo(**{'a': 'a', 'b': 'b'})
a b

In [3]: foo(**{'b': 'b'})
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-c1546e6b8c27> in <module>()
----> 1 foo(**{'b': 'b'})

TypeError: foo() missing 1 required positional argument: 'a'
```

Reply:
Yeah, I hadn't realised you could pass positional parameters in a **argument. That's done now.